lighthouse-core/audits/uses-rel-preconnect.js

```diff
@@ -135,6 +140,12 @@ class UsesRelPreconnectAudit extends Audit {
     const preconnectLinks = artifacts.LinkElements.filter(el => el.rel === 'preconnect');
     const preconnectOrigins = new Set(preconnectLinks.map(link => URL.getOrigin(link.href || '')));
 
+    if (preconnectLinks.length >= 6) {
```

Comment:
@brendankenny we explicitly avoided doing this in the original PR for this audit because we don't know which origin is actually the most impactful one for the UX to be preconnecting to so we show the users all the options. 

maybe we need to add something more explicit to state how many of the ones below they should be adding, but I would not want to only show `maxAllowedPreconnects - currentPreconnectCount`

Reply:
> we explicitly avoided doing this in the original PR for this audit because we don't know which origin is actually the most impactful one for the UX to be preconnecting to so we show the users all the options.

that's a good point, but it's a bit weird for it to suggest a long list at 5 and then suddenly say nothing at 6 (including if the 6 did nothing to speed things up)

Reply:
if we wanted to defer making a change on that, this could also just push a warning if `preconnectLinks.length > 8` and not do the early return on `preconnectLinks.length >= 6)`, instead proceeding as before

Reply:
Very true. I like the idea of throwing a warning and passing it but still including all the origins that would be other options 👍